lib/sparrow/h2_worker.ex

```diff
@@ -52,16 +53,33 @@ defmodule Sparrow.H2Worker do
       end
 
     state = Sparrow.H2Worker.State.new(nil, config)
+
+    :telemetry.execute(
+      [:sparrow, :h2_worker, :init],
+      %{
+        args: %{
+          config: config
+        }
+      },
+      %{state: state}
+    )
+
     {:ok, state, {:continue, :start_conn_backoff}}
   end
 
   @spec terminate(reason, state) :: :ok
-  def terminate(reason, %State{connection_ref: nil}) do
+  def terminate(reason, state = %State{connection_ref: nil}) do
     _ =
       Logger.info(fn ->
         "action=terminate, reason=#{inspect(reason)}, connection_ref=nil"
       end)
 
+    :telemetry.execute(
+      [:sparrow, :h2_worker, :terminate],
+      %{reason: reason},
```

Comment:
This argument is supposed to be measurements - numeric data that can be fed to metric processor. The 3rd argument should hold all necessary metadata.